### ISSUE TITLE
Inner Joins exits the NodeJs Process

### ISSUE THREAD

**Issue by rayishome** [duckdb/duckdb#5370]
### What happens?

This query exits the node process: Any inner join
	
	select d.dimvalue, count(*) from Fact f inner join dimTable d on d.did = f.did group by d.dimvalue

	This query returns no values showing no missing values, but even if there where it shouldn't be an issue:

	select * from Fact f where f.did not in (select did from dimTable)

	This query works, but an inner join is needed for more advanced queries:
	
	select d.dimvalue, count(*) from Fact f left join dimTable d on d.did = f.did group by d.dimvalue

### To Reproduce

The query above should reproduce the issue.  It's an issue in 5.1 and 6.0.  My dataset has large Varchar columns so it's possible it's running out of memory, but not sure since the processes exists without an error code.  

### OS:

x64

### DuckDB Version:

5.1 and 6.0

### DuckDB Client:

NodeJS

### Full Name:

Ray Malone

### Affiliation:

No affiliation

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mause** [comment]
Are you able to provide a reproduction of your NodeJS code as well please?

**Comment by rayishome** [comment]
@Mause  I changed a few lines, but it's mostly intact.  Hope this helps.  I've tested this with Node 14 and 18.3  This is another issue post that looks to be the same: https://github.com/duckdb/duckdb/issues/5344

`const dbexec = async (_db,query,_callback,params) =>{
  let waiter = {}
  let runPromise = new Promise(function(resolve, reject){
    waiter.resolve  = resolve;
    waiter.reject   = reject;
  })
  const whenDone = (err)=> {
    if (_callback){
      _callback(err).then(e=> waiter.resolve());
    }else{
      waiter.resolve(err)
    }
    
    
  }
  _db.exec(query,whenDone);
  return runPromise;

}

const dball = async (_db,query,_callback,params) =>{
  let waiter = {}
  let runPromise = new Promise(function(resolve, reject){
    waiter.resolve  = resolve;
    waiter.reject   = reject;
  })
  const rtn = {status:true,err:false,results:false}
  const whenDone = (err, res)=> {
    if (_callback){
      _callback(err,res).then(e=> {
        if (err){
          rtn.status = false;
          rtn.err = err;
          
          
        } else{
          rtn.results = e;
        }

        waiter.resolve(rtn)
        });
    }else{
      if (err){
        rtn.status = false;
        rtn.err = err;
        
        
      } else{
        rtn.results = res;
      }

      waiter.resolve(rtn)
    }
    
    
  }
  _db.all(query,whenDone);
  return runPromise;

}
`
`var QueryStr = `select d.dimvalue, count(*) from Fact f inner join dimTable d on d.did = f.did group by d.dimvalue`
  
  try{
  var startTime = Date.now();
  var lastTime = Date.now();
  var timedEvents = [];
  
  const logTime = async (eventName)=>{
      const millisl = Date.now() - lastTime;
      const secondsl = Math.floor(millisl / 1000);
      const milliss = Date.now() - startTime;
      const secondss = Math.floor(milliss / 1000);
      timedEvents.push({eventName:eventName,fromLast: secondsl, fromStart:secondss});
      console.log({eventName:eventName,fromLast: secondsl, fromStart:secondss});
      lastTime=Date.now();
      
  }

  const logResponse = (_area,_onlyShowError) =>{
    const area = _area;
    const onlyShowError = _onlyShowError;
    const _logResponse = async (err,results) =>{
      if (onlyShowError){
        if (err) console.log(area,err);
      }else{
        console.log(area,err);
        await logTime(area);
      }
      
      
    }
    return _logResponse;
  }
  
  const sendResponse = async (err,results) =>{
    
    logResponse("send response")();    
    res.send(results);
  }

  await logTime("start");
    
  var duckdb = require('duckdb');
  const path = require("path");
  
  var fileDir = '/export/';
  var fileRoot = global.appRoot;


    
  var fileNewDir = `${fileDir}`;
  const fs = require('fs');  
  const databaseDir = path.join(fileRoot,fileNewDir);
  
  var db = new duckdb.Database(':memory:'); // or a file name for a persistent DB
  await dbexec(db,`SET memory_limit='10GB'`,logResponse("set memory"));
  await dbexec(db,`IMPORT DATABASE '${databaseDir}'`,logResponse(`IMPORT DATABASE '${databaseDir}`));    
  
  await dball(db,QueryStr,sendResponse)
}catch (err) {
  res.send(err);
}
});`

**Comment by Mause** [comment]
Which operating system and which version thereof are you using? You only included your architecture unfortunately

**Comment by rayishome** [comment]
> Which operating system and which version thereof are you using? You only included your architecture unfortunately

Windows 10 and Windows Server 2016 - Tested on both.

**Comment by mazito** [comment]
Any workarounds for these ? 
I am blocked by this issue, as it turns Duckdb almost useless. 
Thanks

**Comment by Alex-Monahan** [comment]
> Any workarounds for these ? I am blocked by this issue, as it turns Duckdb almost useless. Thanks

Unfortunately I am not experienced enough with the node client to know the root cause. As far as workarounds, have you tried using version 0.6.1? If you are on Windows, could you try using WSL to see if it is an OS issue?

In the worst case, it sounds like left joins are working, so you could likely re-write things as left joins and then add in a where clause like:
```sql
WHERE
    table_on_right.join_criteria_one is not null
    and table_on_right.join_criteria_two is not null
```

**Comment by hughess** [comment]
We've just run into this issue as well - on duckdb version 0.6.1.

The following query works on Mac, but exits the node process on Windows:
```
select item 
from orders 
join deliveries on orders.id = deliveries.order_id
```

Making this a left join works on Windows:
```
select item 
from orders 
left join deliveries on orders.id = deliveries.order_id
```

But adding in a where clause causes it to break again:
```
select item 
from orders 
left join deliveries on orders.id = deliveries.order_id
where deliveries.order_id not in ('abc', 'def')
```

**Comment by carlopi** [comment]
This seems to me a duplicate of #5344, could @hughess, @mazito or @rayishome check again either against duckdb 0.7.1 OR duckdb@next?

Thanks a lot, closing as I think has been sorted out pre-0.7.1 release.
